ush/python/pygfs/task/marine_analysis.py

```diff
@@ -0,0 +1,490 @@
+        """
+
+        # prepare background list for the pseudo model, check bkg date for consistency
+        mdau.gen_bkg_list(bkg_path='./bkg',
+                          window_begin=self.task_config.MARINE_WINDOW_BEGIN,
+                          yaml_name='bkg_list.yaml')
+
+        # Make a copy of the env config before modifying to avoid breaking something else
+        envconfig_jcb = copy.deepcopy(self.task_config)
+        logger.info(f"---------------- Prepare the yaml configuration")
+        logger.info(f"{envconfig_jcb}")       # Prepare the yaml configuration
+
+        # Add the things to the envconfig in order to template JCB files
+        envconfig_jcb['PARMgfs'] = self.task_config.PARMgfs
+        envconfig_jcb['nmem_ens'] = self.task_config.NMEM_ENS
+        envconfig_jcb['berror_model'] = 'marine_background_error_static_diffusion'
+        if self.task_config.NMEM_ENS > 3:
+            envconfig_jcb['berror_model'] = 'marine_background_error_hybrid_diffusion_diffusion'
+        envconfig_jcb['DATA'] = self.task_config.DATA
+        envconfig_jcb['OPREFIX'] = self.task_config.OPREFIX
+        envconfig_jcb['PDY'] = os.getenv('PDY')
+        envconfig_jcb['cyc'] = os.getenv('cyc')
+        envconfig_jcb['SOCA_NINNER'] = self.task_config.SOCA_NINNER
+        envconfig_jcb['obs_list'] = ['adt_rads_all']
+
+        # Write obs_list_short
+        with open('obs_list_short.yaml', 'w') as file:
+            yaml.dump(parse_obs_list_file(self.task_config.MARINE_OBS_LIST_YAML), file, default_flow_style=False)
+        os.environ['OBS_LIST_SHORT'] = 'obs_list_short.yaml'
+
+        # Render the JCB configuration files
+        jcb_base_yaml = os.path.join(self.task_config.PARMsoca, 'marine-jcb-base.yaml')
+        jcb_algo_yaml = os.path.join(self.task_config.PARMsoca, 'marine-jcb-3dfgat.yaml.j2')
+
+        jcb_base_config = YAMLFile(path=jcb_base_yaml)
+        jcb_base_config = Template.substitute_structure(jcb_base_config, TemplateConstants.DOUBLE_CURLY_BRACES, envconfig_jcb.get)
+        jcb_base_config = Template.substitute_structure(jcb_base_config, TemplateConstants.DOLLAR_PARENTHESES, envconfig_jcb.get)
+        jcb_algo_config = YAMLFile(path=jcb_algo_yaml)
+        jcb_algo_config = Template.substitute_structure(jcb_algo_config, TemplateConstants.DOUBLE_CURLY_BRACES, envconfig_jcb.get)
+        jcb_algo_config = Template.substitute_structure(jcb_algo_config, TemplateConstants.DOLLAR_PARENTHESES, envconfig_jcb.get)
+
+        # Override base with the application specific config
+        jcb_config = {**jcb_base_config, **jcb_algo_config}
+
+        # convert datetime to string
+        jcb_config['window_begin'] = self.task_config.MARINE_WINDOW_BEGIN.strftime('%Y-%m-%dT%H:%M:%SZ')
+        jcb_config['window_middle'] = self.task_config.MARINE_WINDOW_MIDDLE.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        # Render the full JEDI configuration file using JCB
+        jedi_config = render(jcb_config)
```

Comment:
Not sure why this is being done here and using JCB. Please look at the other scripts using JCB

Reply:
copy/paste from what @danholdaway did. I'll make an issue of this. A lot of that refactoring work is going to be yet again refactored by @DavidNew-NOAA , it can be addressed then.

Reply:
https://github.com/NOAA-EMC/GDASApp/issues/1296